src-front/src/components/WordCloud/WordCloud.tsx

```diff
@@ -0,0 +1,241 @@
+        pixiText: new PIXI.Text({ text: "word5" }),
+        mass: 14,
+      },
+      {
+        text: "",
+        position: new Vector(300, 600),
+        speed: new Vector(0, 0),
+        pixiText: new PIXI.Text({ text: "word6" }),
+        mass: 15,
+      },
+    ];
+
+    refWords.current.forEach((word) => {
+      word.pixiText.style = {
+        fontFamily: "Arial",
+        fontSize: 24,
+        fill: 0xffffff,
+      };
+      word.pixiText.position.set(word.position.x, word.position.y);
+      app.stage.addChild(word.pixiText);
+    });
+
+    //render loop
+    app.ticker.add((time) => {
+      refWords.current.forEach((word) => {
+        //update the speed of words
+        word = updateSpeed(word);
+
+        //update the position of words
+        word.position = word.position.add(
+          word.speed.mult(time.deltaTime * 0.01),
+        );
+
+        word.pixiText.position.set(word.position.x, word.position.y);
+      });
+    });
+
+    return app;
+  }, [containerSize, updateSpeed]);
+
+  //cleanup function that erases the canvas when the page unmounts
+  useEffect(() => {
+    const app = init();
+    const currentRefCanvas = refCanvas.current;
+
+    return () => {
+      app.then((appV) => {
+        appV.destroy();
+        if (currentRefCanvas) {
+          const canvas = currentRefCanvas.querySelector("#pixi-canvas");
+          if (canvas) {
+            currentRefCanvas.removeChild(canvas);
+          }
+        }
+      });
+    };
+  }, [init]);
+
+  //function that updates the size of the container
+  useEffect(() => {
+    const updateSize = () => {
+      //updates the size according to the window
+      if (refContainer.current) {
+        setContainerSize({
+          width: refContainer.current.clientWidth,
+          height: refContainer.current.clientHeight,
+        });
+      }
+    };
+    window.addEventListener("resize", updateSize); //the listener calls the previous function
+    updateSize();
+
+    return () => window.removeEventListener("resize", updateSize);
+  }, []);
+
+  //function that tracks the keypresses
+  const handleKeyPress = useCallback(
+    (event: KeyboardEvent) => {
+      if (event.code === "Enter") {
+        for (let i = 0; i < refWords.current.length; i++) {
+          const word = refWords.current[i];
+          if (word.pixiText.text === inputValue) {
+            deleteWord(i);
+            setInputValue("");
+          }
+        }
+      }
+    },
+    [inputValue],
+  );
+
+  useEffect(() => {
+    window.addEventListener("keydown", handleKeyPress);
+
+    return () => {
+      window.removeEventListener("keydown", handleKeyPress);
+    };
+  }, [handleKeyPress]);
+
+  return (
+    <>
+      <div
+        ref={refContainer}
+        style={{ width: "700px", height: "700px", position: "relative" }}
+      >
+        <div ref={refCanvas}></div>
+        <input
+          type="text"
+          value={inputValue}
+          onChange={handleInputChange}
```

Comment:
Any reason for the usage of `onchange` over `oninput`? If we want to keep track of word progression, `oninput` will be needed I think.